packages/mui-material/src/TextField/TextField.js

```diff
@@ -147,6 +147,15 @@ const TextField = React.forwardRef(function TextField(inProps, ref) {
     InputMore['aria-describedby'] = undefined;
   }
 
+  const handleClick = (event) => {
+    if (!disabled && onClick) {
+      // The `onClick` is registered both on the root and the input elements.
+      // Without stopping the propagation, the event could be triggered twice.
+      event.stopPropagation();
```

Comment:
We can't stop propagation, the click listener could be higher in the DOM tree. I think it should be reverted.